### PR TITLE
Update to latest version of terraform and terraform plugins and use a dynamic vcl block to reduce the amount of repetition in the terraform config

### PR DESCRIPTION
**Pull request by JakeChampion**
When working on this project I noticed that the versions of terraform and the terraform fastly prodiver plugin were quite old.

I've updated the project to use the latest versions and have made use of a new terraform feature known as dynamic blocks to help reduce on some repetition when defining the vcl files.

### CODE REVIEW

File: fastly/terraform/.terraform.lock.hcl [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# This file is maintained automatically by "terraform init".
```
Comment (JakeChampion) [inline fastly/terraform/.terraform.lock.hcl:1]: This is a lockfile, the same concept as npm's `package-lock.json` but for terraform, it ensures we use the same versions of the fastly provider plugin

**Comment by notlee** [comment]
This looks good to me :) Does the terraform version need to be updated in the workflows?
https://github.com/Financial-Times/polyfill-service/blob/c116019a94b41be44b5bc933d26285f9e199613f/.github/workflows/deploy-to-production.yml#L6

**Review by notlee — APPROVED** [review]

File: fastly/terraform/.terraform.lock.hcl (hunk 1 above)
Comment (alexwilson) [reply fastly/terraform/.terraform.lock.hcl:1]: Nice!

File: fastly/terraform/main.tf [hunk 2]
```diff
@@ -1,65 +1,31 @@
...
-
-  vcl {
-    name    = "top_pops.vcl"
-    content = file("${path.module}/../vcl/top_pops.vcl")
+  dynamic "vcl" {
+    for_each = local.vcl
+    content {
+      name    = vcl.value
+      main    = vcl.value == "main.vcl" ? true : false
+      content = templatefile("${path.module}/vcl/${vcl.value}", {})
```
Comment (alexwilson) [inline fastly/terraform/main.tf]: You *might* not need templatefile here, but I guess it's handy!

File: .github/workflows/deploy-to-production.yml [hunk 3]
```diff
@@ -3,7 +3,7 @@ on:
   release:
     types: [created]
 env:
-  terraform_version: '0.12.29'
+  terraform_version: '0.14.8'
```
Comment (alexwilson) [inline .github/workflows/deploy-to-production.yml:6]: Nice, guess this plus the lockfile guarantees that things are updated together?

**Review by alexwilson — APPROVED** [review]
Looks good, and a nice reduction in code!

File: .github/workflows/deploy-to-production.yml (hunk 3 above)
Comment (JakeChampion) [reply .github/workflows/deploy-to-production.yml:6]: Exactly :-D

File: fastly/terraform/main.tf (hunk 2 above)
Comment (JakeChampion) [reply fastly/terraform/main.tf]: Yes, we don't really need it as we are not providing any variables to the vcl files right now but I figured I'd keep it around to keep it similar to other Fastly configs in the company.

**Pull request closed without merging**